tests/JettonMaster.spec.ts

```diff
@@ -299,14 +301,18 @@ describe('JettonMaster', () => {
             to: jettonMaster.address,
             success: true,
             deploy: false,
-            op: 0x133703,
+            op: OP_CODES.JettonSetParameter,
         });
 
         // Checks
         let jettonMasterMetadata = await jettonMaster.getGetJettonData();
-        expect(jettonMasterMetadata.mintable).toEqual(false);
+        //expect(jettonMasterMetadata.mintable).toEqual(false); 
+        // Should max supply be allowed to be set to 0 so mintable will be false?
```

Comment:
This is what mostly got my attention for this PR. Some answers are hard to find for TON , and using `max_supply` for controlling the  `mintable` didnt make much sense to me.   
`ERROR_CODE_INVALID_AMOUNT` was mostly added for gas opt and prevent any unexpected behavior.

Thanks for getting back on this with clear answers!

Reply:
You are right, managing `mintable` via `max_supply` is implicit behavior. There should be an explicit way to stop minting new tokens. Not via limiting supply
I thought about it for a while and realized that `max_supply` field is not really needed and can be replaced with the `mintable` flag. Especially considering that `max_supply` is 256 bits and `mintable` is only 1.
I will open issue for it

Reply:
I will remove the added logic for `max_supply` from this PR so this change can be on its own on a separate ticket. I'll also correct my git commit wording (i have a bad habit of using "add" instead of "feat", apologies) and submit the changes for another review